### PR TITLE
Stop passing in loggers to clusters explicitly

### PR DESCRIPTION
**Pull request by ncabatoff**
The goal here is to rely instead on the TestLogger that gets created when a logger isn't passed in.  This makes the tests shorter, but more importantly will facilitate some other code that wants the cluster's logger to be an InterceptLogger.  And by virtue of using TestLogger, we ensure that after Cleanup gets called, we don't get any more test output from logging, which is almost always noise.

### CODE REVIEW

File: command/agent_test.go [hunk 1]
```diff
@@ -2624,10 +2615,7 @@ func TestAgent_Metrics(t *testing.T) {
 
 	// Start a vault server
 	logger := logging.NewVaultLogger(hclog.Trace)
```
Comment (mpalmi) [inline command/agent_test.go:2617]: We can probably move this down to the `testAgentCommand` invocation so it's more apparent what it's for.

File: command/agentproxyshared/auth/auth_test.go [hunk 2]
```diff
@@ -60,7 +60,6 @@ func (u *userpassTestMethod) Shutdown() {
 func TestAuthHandler(t *testing.T) {
 	logger := logging.NewVaultLogger(hclog.Trace)
```
Comment (mpalmi) [inline command/agentproxyshared/auth/auth_test.go:61]: Should we use the `TestLogger` for `NewAuthHandler` below, or do we want to stick with the `NewVaultLogger`

File: command/agent_test.go (hunk 1 above)
Comment (ncabatoff) [reply command/agent_test.go]: Done.

File: command/agentproxyshared/auth/auth_test.go (hunk 2 above)
Comment (ncabatoff) [reply command/agentproxyshared/auth/auth_test.go]: I chose the latter.

File: command/agent/alicloud_end_to_end_test.go [hunk 3]
```diff
@@ -49,9 +49,8 @@ func TestAliCloudEndToEnd(t *testing.T) {
 	}
 	testhelpers.SkipUnlessEnvVarsSet(t, credNames)
 
-	logger := logging.NewVaultLogger(hclog.Trace)
+	logger := logging.NewVaultLogger(log.Trace)
```
Comment (mpalmi) [inline command/agent/alicloud_end_to_end_test.go:52]: Can we drop this and use `cluster.Logger` below?

**Review by mpalmi — APPROVED** [review]

**Pull request merged**